src/Numerics/DGMethods/FVModel.jl

```diff
@@ -0,0 +1,483 @@
+using StaticArrays
+
+using ClimateMachine
+using ClimateMachine.VariableTemplates
+
+
+"""
+Fifth order WENO reconstruction on nonuniform grids
+Implemented based on 
+Wang, Rong, Hui Feng, and Raymond J. Spiteri. 
+"Observations on the fifth-order WENO method with non-uniform meshes." 
+Applied Mathematics and Computation 196.1 (2008): 433-447.
+
+size(Δh) = 5
+size(u) = (num_state_primitive, 5)
+construct left/right face states of cell[3]
+h1     h2     h3     h4      h5
+|--i-2--|--i-1--|--i--|--i+1--|--i+2--|
+without hat : i - 1/2
+with hat    : i + 1/2  
+r = 0, 1, 2 cells to the left, => P_r^{i}
+I_{i-r}, I_{i-r+1}, I_{i-r+2}
+P_r(x) =  ∑_{j=0}^{2} C_{rj}(x) u_{i - r + j}  (use these 3 cell averaged values)
+C_{rj}(x) = B_{rj}(x) h_{3-r+j}                (i = 3)
+C''_{rj}(x) = B''_{rj}(x) h_{3-r+j}                (i = 3)
+P''_r(x)    =  ∑_{j=0}^{2} C''_{rj}(x) u_{i - r + j}  (use these 3 cell averaged values)
+=  ∑_{j=0}^{2} B''_{rj}(x) h_{3-r+j}  u_{i - r + j}  (use these 3 cell averaged values)
+
+"""
+
+function weno_reconstruction!(
+    state_primitive_top::Vars,
+    state_primitive_bottom::Vars,
+    cell_states_primitive::NTuple{5, Vars},
+    cell_weights::SVector{5, FT},
+) where {FT}
+
+
+    num_state_primitive = length(parent(state_primitive_top))
+    h1, h2, h3, h4, h5 = cell_weights
+
+    b̂ = zeros(FT, 3, 3)
```

Comment:
```suggestion
    b̂ = similar(parent(state_primitive_top), 3, 3)
```
`zeros` will return an array that can only reside on the CPU, so let’s use similar (which should be an MArray). Might need to check the syntax, `similar` might need tuple instead of multiple args